### DISCUSSION TITLE
Internal temperature error ?

### DISCUSSION THREAD

**Discussion by jopl** [deividAlfa/stm32_soldering_iron_controller#109] [Q&A]
Hi Deivid,
just a simple question: can I calibrate somehow the internal temperature, please? It starts at 28.8 °C after power on and it is rapidly increasing to approx. 35-40 °C after that despite the room temperature around 23 °C. Or maybe it is normal state, but microcontroller seems to be cold ... I can try to measure it, if it is needed. Thx for reply. BTW, it is amazing firmware! Thank you for it!!! Josef.

**Comment by deividAlfa** [comment]
No, this is the way it should work, it's not the ambient temperature but the sensor cold function.

**Comment by jopl** [comment]
Dear deividAlfa,
thanks, I know, but 40 °C seems to me relatively high, when I cannot feel the microcontroller as warm, you know. I finally also saw the soldering station KSGER, which shows temperature very close to ambient ... this is why I asked about modifying that.
But OK, if it is normal, no problem with that.
Regards, Josef.

**Reply by deividAlfa** [reply]
Of course 40°C is slightly warm to touch? You're at 36°C so it's just slightly hotter. You won't feel burning until 50°C or so.

Internally the handle will get hotter than that, so you're already inducing some error by not having the ntc there.
I repeat , "fixing" it will only increase the tip error.

**Comment by jopl** [comment]
OK, thank you for your expertize. I added the NTC to the G12 connector internally and tried to solder. It solved the situation (the tip is not changing temperature from this time ;-)). I know the best solution is to have NTC in the handle, but unfortunately I have got just 4 wires ...
Regards, Josef.
